tests/app/dao/test_jobs_dao.py

```diff
@@ -409,3 +413,105 @@ def test_can_letter_job_be_cancelled_returns_false_and_error_message_if_notifica
     result, errors = can_letter_job_be_cancelled(job)
     assert not result
     assert errors == "We are still processing these letters, please try again in a minute."
+
+
+def test_find_jobs_with_missing_rows(sample_email_template):
```

Comment:
in this test we could also create a job that is finished and not missing any rows, and then show that `find_jobs_with_missing_rows` is only finding the job with missing rows, but not the "healthy" job.

Reply:
Good shout! there's a bug with the group by